graphics/macgui/macdialog.cpp

```diff
@@ -116,97 +115,61 @@ void Dialog::paint() {
 
 		if ((int)i == _pressedButton && _mouseOverPressedButton) {
 			Common::Rect bb(button->bounds.left + 5, button->bounds.top + 5,
-				button->bounds.right - 5, button->bounds.bottom - 5);
+							button->bounds.right - 5, button->bounds.bottom - 5);
 
-			Design::drawFilledRect(&_gui->_screen, bb, kColorBlack, _gui->_wm->getPatterns(), kPatternSolid);
+			drawFilledRect1(bb, kColorBlack, _wm->getDrawPixel(), &pd);
 
 			color = kColorWhite;
 		}
 		int w = _font->getStringWidth(button->text);
 		int x = button->bounds.left + (button->bounds.width() - w) / 2;
 		int y = button->bounds.top + 6;
 
-		_font->drawString(&_gui->_screen, button->text, x, y, _bbox.width(), color);
+		_font->drawString(_screen, button->text, x, y, _bbox.width(), color);
 
 		drawOutline(button->bounds, buttonOutline, ARRAYSIZE(buttonOutline));
 	}
 
-	g_system->copyRectToScreen(_gui->_screen.getBasePtr(_bbox.left, _bbox.top), _gui->_screen.pitch,
-			_bbox.left, _bbox.top, _bbox.width() + 1, _bbox.height() + 1);
+	g_system->copyRectToScreen(_screen->getBasePtr(_bbox.left, _bbox.top), _screen->pitch,
+							   _bbox.left, _bbox.top, _bbox.width() + 1, _bbox.height() + 1);
 
 	_needsRedraw = false;
 }
 
-void Dialog::drawOutline(Common::Rect &bounds, int *spec, int speclen) {
+void MacDialog::drawOutline(Common::Rect &bounds, int *spec, int speclen) {
+	MacPlotData pd(_screen, nullptr, &_wm->getPatterns(), 1, 0, 0, 1, _wm->_colorBlack, false);	
 	for (int i = 0; i < speclen; i++)
-		if (spec[i] != 0)
-			Design::drawRect(&_gui->_screen, bounds.left + i, bounds.top + i, bounds.right - i, bounds.bottom - i,
-						1, kColorBlack, _gui->_wm->getPatterns(), kPatternSolid);
-}
-
-int Dialog::run() {
```

Comment:
I wonder, why is this running on an engine level but is not here? You may just return different things like "should quit" if EVEN_QUIT is set and kDialogOption1/kDialogOption2 for the requested options